pkg/apiserver/handlers_test.go

```diff
@@ -200,6 +200,35 @@ func TestTimeout(t *testing.T) {
 	}
 }
 
+func TestHasApiPrefix(t *testing.T) {
```

Comment:
These are necessary, but not sufficient.  Enough logic changed in `GetAttribs`, that it requires tests as well.  I'm particular concerned about the current factorization that bases a resource parse on one condition and a `NonResourcePath` on another.

Making that code mutually exclusive will help inspectability, but it will still need tests.